embulk-core/src/main/java/org/embulk/jruby/LazyScriptingContainerDelegate.java

```diff
@@ -18,6 +27,101 @@ public LazyScriptingContainerDelegate(
         this.initializer = initializer;
     }
 
+    public static LazyScriptingContainerDelegate withGems(
+            final Logger logger,
+            final EmbulkSystemProperties embulkSystemProperties) {
+        return of(false, true, null, logger, embulkSystemProperties);
+    }
+
+    public static LazyScriptingContainerDelegate withGemsIgnored(
+            final Logger logger,
+            final EmbulkSystemProperties embulkSystemProperties) {
+        return of(false, false, null, logger, embulkSystemProperties);
+    }
+
+    public static LazyScriptingContainerDelegate withInjector(
+            final Injector injector,
+            final Logger logger,
+            final EmbulkSystemProperties embulkSystemProperties) {
+        return of(true, true, injector, logger, embulkSystemProperties);
+    }
+
+    private static LazyScriptingContainerDelegate of(
+            final boolean isEmbulkSpecific,
+            final boolean initializesGem,
+            final Injector injector,
+            final Logger logger,
+            final EmbulkSystemProperties embulkSystemProperties) {
+        // use_global_ruby_runtime is valid only when it's guaranteed that just one Injector is
+        // instantiated in this JVM.
+        final boolean useGlobalRubyRuntime = embulkSystemProperties.getPropertyAsBoolean("use_global_ruby_runtime", false);
+
+        final String jrubyProperty = embulkSystemProperties.getProperty("jruby");
+
+        final ArrayList<URL> jrubyUrlsBuilt = new ArrayList<>();
+        if (jrubyProperty != null && !jrubyProperty.isEmpty()) {
+            // File.pathSeparator is not available here because the property "jruby" expects a URL-like style:
+            // "file:" and "mvn:". It now supports only "file:", though.
+            //
+            // Semicolons basically do not appear in URLs without quoting except for the "optional fields and values"
+            // case described in RFC 1738. We don't need to take care of the "optional fields and values" case here.
+            // https://tools.ietf.org/html/rfc1738
+            for (final String jarLocator : jrubyProperty.split("\\;")) {
+                if (jarLocator.startsWith("file:")) {
+                    // TODO: Validate the path more.
```

Comment:
Is this TODO a leftover on purpose?

Reply:
Ah, not on a very specific purpose, but it was just a possible future improvement for providing a better error message / log for a case that a user configures `jruby` wrongly.

It is now setting user's configuration directly to `URLClassLoader`, so a user may see some cryptic error messages when the configuration is wrong. I thought it's worth validating user's configuration and providing a better error message, but wanted to think about that later...

Reply:
Got it. Not necessary to get everything into this PR, but some TODOs alert me :)